pkg/pipelineascode/match.go

```diff
@@ -59,7 +59,7 @@ func (p *PacRun) matchRepoPR(ctx context.Context) ([]matcher.Match, *v1alpha1.Re
 	// allow webhook providers users to have a global webhook secret to be used,
 	// so instead of having to specify their in Repo each time, they use a
 	// shared one from pac.
-	if repo.Spec.GitProvider != nil && p.event.InstallationID <= 0 {
+	if p.event.InstallationID <= 0 {
```

Comment:
i still cant rem why i made that logic <= 0 i am sure there is a reason for it....

Reply:
for github webhook its -1 and for rest of provider its 0
thats why 😄

Reply:
yeah i understood but that logic can be inverted : 

```go
	if p.event.InstallationID > 0 {
		p.event.Provider.WebhookSecret, _ = GetCurrentNSWebhookSecret(ctx, p.k8int)
	} else {
		err := SecretFromRepository(ctx, p.run, p.k8int, p.vcx.GetConfig(), p.event, repo, p.logger)
		if err != nil {
			return nil, nil, err
		}
	}
```